Add community PR process documentation (#471)
bitwarden/contributing-docs: docs/contributing/pull-requests/community-pr-process.md

---
sidebar_position: 4
sidebar_custom_props:
  access: community
---

# Contribution Review Process

Congratulations! You've submitted a pull request for changes to the Bitwarden codebase.

Bitwarden recognizes the time and effort that community members take to recognize bugs or potential
new features and offer up pull requests to resolve them.

Transparency is one of our core values, and we feel that it is important that contributors
understand the process that our teams take in evaluating, prioritizing, and reviewing these pull
requests, and why some changes may move more quickly to completion than others.

## Product assessment

When your community contribution is received, the first step is for our Product team to evaluate

> whether the change aligns with the vision for the Bitwarden product. This is more relevant for

Reviewer: ```suggestion
whether the change aligns with the vision for the Bitwarden products. This is more relevant for
```

Author: Addressed with [`539ba43` (#471)](https://github.com/bitwarden/contributing-docs/pull/471/commits/539ba437c455b0df90d2c3b02dda0681cb87868d).

feature enhancements than bugs, but it is important for us to assess whether the change conflicts
with larger upcoming changes on the roadmap or would potentially replace work already on the backlog
for a team.

If the Product team has clarifying questions or would like the change to be modified to more closely
align with their vision, they will reach out on the PR and request changes.

Once Product has signed off on the change, it moves to our Engineering teams.

## Engineering assessment

The Engineering team(s) responsible for the change depend largely on what code was changed. We use

> Github code owners to manage the areas in our repositories owned by each team, and the responsible

Reviewer: ```suggestion
GitHub code owners to manage the areas in our repositories owned by each team, and the responsible
```

Author: [`539ba43` (#471)](https://github.com/bitwarden/contributing-docs/pull/471/commits/539ba437c455b0df90d2c3b02dda0681cb87868d)

team(s) will be automatically tagged as reviewers on the PR that you raise to merge your changes.

:::tip Assigned reviewers

When a Bitwarden developer is assigned to your PR, that does not necessarily mean that the PR is
under active review at that moment. This assignment may take place while the work is still on the
team's backlog.

:::

> The responsible Engineering teams will review the change and ensure that there are no concerns

Reviewer: ```suggestion
The responsible Engineering team(s) will review the change and ensure that there are no concerns
```

Author: Addressed with [`539ba43` (#471)](https://github.com/bitwarden/contributing-docs/pull/471/commits/539ba437c455b0df90d2c3b02dda0681cb87868d).

> around how the change was implemented. This is generally more applicable for larger sets of changes,

Reviewer: ```suggestion
around how the change was implemented; this is generally more applicable for larger sets of changes,
```

Author: Addressed with [`539ba43` (#471)](https://github.com/bitwarden/contributing-docs/pull/471/commits/539ba437c455b0df90d2c3b02dda0681cb87868d).
